aws/rust-runtime/aws-config/src/lib.rs

```diff
@@ -238,14 +239,14 @@ mod loader {
     use crate::provider_config::ProviderConfig;
 
     #[derive(Default, Debug)]
-    enum CredentialsProviderOption {
-        /// No provider was set by the user. We can set up the default credentials provider chain.
+    enum TriStateOption<T> {
```

Comment:
We used to have a TriState in aws-smithy-types. You could consider resurrecting that if you want to, but doesn't matter to me either way since this is private.

Reply:
If we have need for another tri-state then we can move it but for now I think I'd keep it private.